source/CleoBase.cpp

```diff
@@ -252,9 +252,9 @@ namespace CLEO
         GetInstance().RemoveCallback(id, func);
     }
 
-    DWORD WINAPI CLEO_GetInternalAudioStream(CLEO::CRunningScript* thread, DWORD stream) // arg CAudioStream *
+    DWORD WINAPI CLEO_GetInternalAudioStream(CLEO::CRunningScript* unused, DWORD audioStreamPtr)
     {
-        return stream; // CAudioStream::streamInternal offset is 0
+        return audioStreamPtr + 0x4; // offsetof(CAudioStream, streamInternal)
```

Comment:
check what you return from here. I think you should dereference the pointer:

```
    DWORD WINAPI CLEO_GetInternalAudioStream(CLEO::CRunningScript* unused, DWORD audioStreamPtr)
    {
        return *(DWORD*)(audioStreamPtr + 0x4);
    }
```

Reply:
True. Fixed.